pkg/controller/daemon/daemoncontroller.go

```diff
@@ -317,8 +317,14 @@ func (dsc *DaemonSetsController) addPod(obj interface{}) {
 // up. If the labels of the pod have changed we need to awaken both the old
 // and new set. old and cur must be *v1.Pod types.
 func (dsc *DaemonSetsController) updatePod(old, cur interface{}) {
-	curPod := cur.(*v1.Pod)
-	oldPod := old.(*v1.Pod)
+	curPod, ok := cur.(*v1.Pod)
+	if !ok {
+		return
```

Comment:
please also log error message if it's failed. ignore is worse than panic, it's hard to admin/dev to investigate issue if any.